### ISSUE TITLE
Documentation for customization

### ISSUE THREAD

**Issue by redfrost** [roylee0704/react-flexbox-grid#87]
Hi, thanks for the great grid system. It's very nice to use right out of the box, but it's hard to figure out where to start to modify some settings, for example gutter size between boxes. 

It would be nice to have a detailed instruction or more examples.

**Comment by silvenon** [comment]
I'm not sure if this feature would be within the scope of this library. It is meant to only be a React wrapper around [flexboxgrid](http://flexboxgrid.com/), which doesn't offer this kind of customization.

I see why people would want to customize it, but currently I don't have any idea how to achieve this. Does anybody else?

[flexboxgrid/src/css/flexboxgrid.css](https://github.com/kristoferjoseph/flexboxgrid/blob/master/src/css/flexboxgrid.css) uses variables, maybe we could somehow use that, but it sounds very tricky.

**Comment by redfrost** [comment]
Thanks for your reply. 
In my case I got a simple `sass` setup like this to override default values, but I was wondering is there any way to approach in raw level in react by hijacking original variables, also I would like to see how other people do it.

```sass
$container-width: 1280px;
$container-gutter: 10px;
$breakpoint-sm: 960px;   // default 768px
$breakpoint-md: 1024px; // default 1024px
$breakpoint-lg: 1280px;   // default 1280px
$breakpoint-xlg:1920px;  // default 1920px

.grid {
  max-width: $container-width;
  padding-right: $container-gutter/2; 
  padding-left:  $container-gutter/2; 
  
    .col {
      padding: $container-gutter;
      @media (max-width: $breakpoint-sm - 1px) {
        max-width: 100%; 
        flex-basis:100%;
      }
    }
}
```

**Comment by silvenon** [comment]
Yeah, this is probably a pain, because you have to understand how `flexboxgrid` is written.

**Comment by silvenon** [comment]
I think CSS Modules support complicate our lives a bit, it would probably be much easier otherwise.

**Comment by alexlacasa86** [comment]
Hi!

I faced this issue recently, and I was able to override the variables using `webpack.NormalModuleReplacementPlugin` like this:

```
/* webpack.config.js */

const NormalModuleReplacementPlugin = require('webpack');

// ...
plugins: [
      new NormalModuleReplacementPlugin(
      /node_modules\/flexboxgrid\/dist\/flexboxgrid\.css/,
      require.resolve('./src/styles/custom-flex-box-grid.css')
    )
]
// ...
```
And then, in that custom file:

```
/* custom-flex-box-grid.css */

@import 'flexboxgrid/src/css/flexboxgrid.css';

:root {
  --gutter-width: 0;
}
```

Hope this can be usefull!

**Comment by planetflash** [comment]
Any idea how to do this on the latest version that uses flexboxgrid2?

**Comment by planetflash** [comment]
I also think there is a typo in @alexlacasa86 solution. I can only get it to work with the setting in webpack.config.js to
```
new webpack.NormalModuleReplacementPlugin(
      /node_modules\/flexboxgrid\/src\/flexboxgrid\.css/,
      require.resolve(path.join(process.cwd(), 'app/sass/custom-flex-box-grid.css'))
    ),
```
Notice `src` in the RegExp not `dist`

**Comment by jsergiu** [comment]
I think one way would be to have more properties on Row and Col to allow more flexibility. I will try to have a look.

**Comment by LaloMrtnz** [comment]
I just created a CSS-in-JS version that can be easily customized, let me know what you think https://github.com/LaloMrtnz/flexgrid-js ;)
